test/weights.jl

```diff
@@ -15,13 +17,33 @@ wv = weights(w)
 @test sum(wv) == 6.0
 @test !isempty(wv)
 
+b  = trues(3)
+bv = weights(b)
+@test eltype(bv) == Bool
+@test length(bv) == 3
+@test values(bv) == b
+@test sum(bv)    == 3
+@test !isempty(bv)
+
+c  = [1, 4, 3, 5]; d = [4, 7, 18, 9]; e = [1, 2, -5, 3];
+s  = sparse(c, d, e)
+cv = weights(c)
+@test eltype(cv) == Int64
+@test length(cv) == 4
+@test values(cv) == c
+@test sum(cv)    == 13.0
```

Comment:
Yeah, `===` makes the test more robust. I would add it to [this line](https://github.com/JuliaStats/StatsBase.jl/pull/152#discussion-diff-56038439L15) too.